planner/core/rule_column_pruning.go

```diff
@@ -220,7 +220,14 @@ func (p *LogicalUnionScan) PruneColumns(parentUsedCols []*expression.Column) err
 func (ds *DataSource) PruneColumns(parentUsedCols []*expression.Column) error {
 	used := expression.GetUsedList(parentUsedCols, ds.schema)
 
-	exprCols := expression.ExtractColumnsFromExpressions(nil, ds.allConds, nil)
+	conds := make([]expression.Expression, len(ds.allConds))
+	copy(conds, ds.allConds)
+	for _, col := range ds.schema.Columns {
```

Comment:
How about ExtractDependentColumns() ?

Reply:
Sure, I'll try it. I was struggling a little bit with the explain test, hopefully this could fix it.